src/matlab2tikz.m

```diff
@@ -638,8 +638,9 @@ function addCustomCode(fid, before, code, after)
     % how MATLAB does it, too. Significant for patch (contour) plots,
     % and the order of plotting the colored patches.
     for child = children(end:-1:1)'
+        m2t = hasLegendEntry(m2t,child);
```

Comment:
Yes, because any nested `handleAllChildren` call might remove the legend status for the parent.